tensorflow/contrib/tensorrt/plugin/trt_plugin.cc

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/contrib/tensorrt/plugin/trt_plugin.h"
+#include <cassert>
+#include <cstring>
+#include "tensorflow/contrib/tensorrt/plugin/trt_plugin_utils.h"
+
+#if GOOGLE_CUDA
+#if GOOGLE_TENSORRT
+
+namespace tensorflow {
+namespace tensorrt {
+
+PluginTensorRT::PluginTensorRT(const void* serialized_data, size_t length) {
+  // sanity check.
+  assert(EncodeOpName(GetPluginName()) !=
```

Comment:
The assert does not really matter here, since it has already been checked by PluginFactoryTensorRT::createPlulgin(...)

I should have used ExtractOpName, since I'm trying to consume the first couple bytes of serializaed_data inserted by the serialize().
I'm doing this simply because of current TRT plugin API. I could wrap it differently to pass serialized_data as (void**) to allow data consumption marked by each deserialize function call.

PluginTensorRT is an abstract class. We could have multiple child-classes utilizing the same serialize() & parent constructor. Each would have different op name that was throw at the serialized_data.